### PR TITLE
fixes realm save

### PR DESCRIPTION
**Pull request by edewit**
currently save on realm screen didn't work as the email settings where required. Fixed this by changing email to be it's own section. Also email section was using the wrong attributes therefor the changes where not reflected in the old console

### CODE REVIEW

**Comment by edewit** [comment]
fixing comment: https://github.com/keycloak/keycloak-admin-ui/issues/607#issuecomment-846941758

File: src/realm-settings/KeysListTab.tsx [hunk 1]
```diff
@@ -2,42 +2,51 @@ import React, { useState } from "react";
...
+    const keysMetaData = await adminClient.realms.getKeys({
+      realm: realmName,
+    });
+    const keys = keysMetaData.keys;
 
-  React.useEffect(() => {
-    refresh();
-  }, [keys]);
+    return keys?.map((key) => {
+      const provider = realmComponents.find(
```
Comment (jenny-s51) [inline src/realm-settings/KeysListTab.tsx:44]: For some reason `realmComponents` is always `undefined` if called in the loader - the providers column of the table is empty when provider is initialized this way. Splitting up into `KeysListTab` into two components resolved this issue and the provider is rendered for each row

**Review by jenny-s51 — CHANGES_REQUESTED** [review]
Save/email form updates look great @edewit ! How did you know to use the `smptServer` object? It looked like the old console was also using `attributes`.

Also, a small thing - if the KeysListTab is consolidated into one component, props passed into the loader are `undefined` when accessed, and the `provider` column is empty for each row if the KeysList. This is why it was split up into two components initially. 

What do you think is the best thing to do here? Should we split it up again or is there a way we can just use the one component?

File: src/realm-settings/KeysListTab.tsx (hunk 1 above)
Comment (edewit) [reply src/realm-settings/KeysListTab.tsx:44]: right it's a timing issue the table gets rendered when the realmComponent array is still empty. Fixed it by making the render depended on the realmComponent array being loaded.

**Review by jenny-s51 — APPROVED** [review]
Thank you for clarifying - updates look awesome 👍

**Pull request merged**